backend/hitas/tests/apis/test_api_condition_of_sale.py

```diff
@@ -0,0 +1,1179 @@
+import uuid
+from typing import Any, NamedTuple
+
+import pytest
+from dateutil.relativedelta import relativedelta
+from django.urls import reverse
+from django.utils import timezone
+from rest_framework import status
+
+from hitas.models import Apartment, ConditionOfSale, Owner, Ownership
+from hitas.models.condition_of_sale import GracePeriod
+from hitas.tests.apis.helpers import HitasAPIClient, InvalidInput, parametrize_helper
+from hitas.tests.factories import (
+    ApartmentFactory,
+    ApartmentSaleFactory,
+    ConditionOfSaleFactory,
+    OwnerFactory,
+    OwnershipFactory,
+)
+from hitas.views.condition_of_sale import ConditionOfSaleSerializer
+
+# List tests
+
+
+@pytest.mark.django_db
+def test__api__condition_of_sale__list__empty(api_client: HitasAPIClient):
+    owner: Owner = OwnerFactory.create()
+
+    url = reverse(
+        "hitas:conditions-of-sale-list",
+        kwargs={
+            "owner_uuid": owner.uuid.hex,
+        },
+    )
+    response = api_client.get(url)
+    assert response.status_code == status.HTTP_200_OK, response.json()
+    assert response.json()["contents"] == []
+    assert response.json()["page"] == {
+        "size": 0,
+        "current_page": 1,
+        "total_items": 0,
+        "total_pages": 1,
+        "links": {
+            "next": None,
+            "previous": None,
+        },
+    }
+
+
+@pytest.mark.django_db
+def test__api__condition_of_sale__list__single(api_client: HitasAPIClient):
+    owner: Owner = OwnerFactory.create()
+    new_ownership: Ownership = OwnershipFactory.create(owner=owner)
+    old_ownership: Ownership = OwnershipFactory.create(owner=owner)
+    condition_of_sale: ConditionOfSale = ConditionOfSaleFactory.create(
+        new_ownership=new_ownership,
+        old_ownership=old_ownership,
+    )
+
+    url = reverse(
+        "hitas:conditions-of-sale-list",
+        kwargs={
+            "owner_uuid": owner.uuid.hex,
+        },
+    )
+    response = api_client.get(url)
+    assert response.status_code == status.HTTP_200_OK, response.json()
+    assert response.json()["contents"] == [
+        {
+            "id": condition_of_sale.uuid.hex,
+            "grace_period": str(condition_of_sale.grace_period.value),
+            "fulfilled": condition_of_sale.fulfilled,
+            "new_ownership": {
+                "id": new_ownership.uuid.hex,
```

Comment:
And here

Reply:
Additional ownerships option on condition of sale creation needs this, see [here](https://github.com/City-of-Helsinki/hitas/pull/212/files#diff-371f597cf9397b67f5f3749ab0f5c08c290e53cf39589034af23ead33edee12bR79).